pep-9999.rst

```diff
@@ -261,15 +261,16 @@ type is assumed to be ``Any``. It is preferred to leave unknown
 types unannotated rather than explicitly marking them as ``Any``, as some
 type checkers can optionally warn about unannotated arguments.
 
-The type of
-an argument where the annotation and the type of the default argument disagree
-is unspecified. As an exception, the ellipsis literal can stand in for any
-type::
-
-    # x has type Any
-    # y's type is unspecified
-    # z has type int
-    def foo(x, y: int = "", z: int = ...): ...
+If an argument has a literal or constant default value, it must match the implementation
+and the type of the argument (if specified) must match the default value.
+Alternatively, ``...`` can be used in place of any default value::
+
+    # The following arguments all have type Any.
+    def unannotated(a, b=42, c=...): ...
+    # The following arguments all have type int.
+    def annotated(a: int, b: int = 42, c: int = ...): ...
+    # The following default values are invalid and the types are unspecified.
```

Comment:
Why would the type be unspecified here, if the type is written out? I know the default value is dubious, but if the type is valid I wouldn't expect a type checker to throw the parameter type away.

Is this just trying to encode the fact that stubs shouldn't be trying to actually "run" anything? I agree that this sort of expression would be best to not appear in a stub, but if a type checker does encounter something like it, the behavior I'd expect is for it to say "okay, it's `Foo`, what you wrote, and maybe I'll tell you that this is a bad idea".

Reply:
PEP 484 used to allow `x: int = None` to mean `Optional[int]` and mypy implemented this. Declaring this as "unspecified" means that a type checker could, for example, error out if it encounters such a discrepancy. What's the benefit in defining a fixed behavior for an invalid construct like this?

Reply:
I see, the I interpreted "unspecified" as "unknown type", and not in the language-spec-y way as "the implementer decides". Thanks for the clarification.

Reply:
Ah I see. We define "unspecified" in the first chapter, so this will hopefully be more clear once the document is complete.

Reply:
Yes, a folly of me reading the diff and not remembering the beginning of the document. 🙂